android/src/main/java/com/lwansbrough/RCTCamera/RCTCameraModule.java

```diff
@@ -162,6 +183,191 @@ public String getName() {
         });
     }
 
+    private boolean prepareMediaRecorder(ReadableMap options) {
+
+        CamcorderProfile cm = RCTCamera.getInstance().setCaptureVideoQuality(options.getInt("type"), options.getString("quality"));
+        
+        mCamera.unlock();  // make available for mediarecorder
+
+        mediaRecorder.setCamera(mCamera);
+
+        mediaRecorder.setAudioSource(MediaRecorder.AudioSource.CAMCORDER);
+        mediaRecorder.setVideoSource(MediaRecorder.VideoSource.CAMERA);
+    
+        int mediaRecorderHintOrientation = (90 + ((720 - RCTCamera.getInstance().getActualDeviceOrientation() * 90))) % 360;
+
+        // adjust for differences in how devices display front facing http://www.theverge.com/2015/11/9/9696774/google-nexus-5x-upside-down-camera
+        if (RCT_CAMERA_TYPE_FRONT == options.getInt("type")) {
+            if ((RCT_CAMERA_ORIENTATION_PORTRAIT == RCTCamera.getInstance().getOrientation()) && (RCTCamera.getInstance().getAdjustedDeviceOrientation() == 90)) {
+                mediaRecorderHintOrientation += 180;
+            } 
+        }
+        mediaRecorder.setOrientationHint(mediaRecorderHintOrientation);
+
+        if (null == cm) {
+            return false; 
+        }
+
+        cm.fileFormat = MediaRecorder.OutputFormat.MPEG_4;
+        cm.videoCodec = MediaRecorder.VideoEncoder.MPEG_4_SP;
+
+        mediaRecorder.setProfile(cm);        
+
+        videoFile = null;
+        switch (options.getInt("target")) {
+            case RCT_CAMERA_CAPTURE_TARGET_MEMORY:
+                videoFile = getTempMediaFile(MEDIA_TYPE_VIDEO); // temporarily
+                break;
+            case RCT_CAMERA_CAPTURE_TARGET_CAMERA_ROLL:
+                videoFile = getOutputMediaFile(MEDIA_TYPE_VIDEO);
+                break;
+            case RCT_CAMERA_CAPTURE_TARGET_TEMP:
+                videoFile = getTempMediaFile(MEDIA_TYPE_VIDEO);
+                break;
+            default:
+            case RCT_CAMERA_CAPTURE_TARGET_DISK:
+                videoFile = getOutputMediaFile(MEDIA_TYPE_VIDEO);
+                break;
+        }
+        if (videoFile == null) {
+            return false;
+        }
+
+        mediaRecorder.setOutputFile(videoFile.getPath());
+
+        // Attach callback for maxDuration and maxFileSize
+        mediaRecorder.setOnInfoListener(this);
+ 
+        if (options.hasKey("maxDuration")) {
```

Comment:
`maxDuration` should be `totalSeconds`, that's what it is called on IOS at least...